modules/internal/ChapelBase.chpl

```diff
@@ -1261,23 +1263,23 @@ module ChapelBase {
   }
 
   inline operator <<(param a: int(?w), param b: integral) param {
-    if boundsChecking then bitshiftChecks(a, b);
+    bitshiftChecks(a, b);
```

Comment:
I can't tell the resolution order off top of my head here. There might be a net negative impact of this with `--no-checks` and `--fast`.

Reply:
I can revert that part of the PR, I think that's less critical to the improvements here than removing `inline`. It just seemed silly to not do these checks with `--no-checks`, since its done entirely at compile time and has no impact on runtime performance

Reply:
> has no impact on runtime performance

I agree about this part, but just uncertain about the impact on compiler performance.

To be clear, I think we don't resolve a block at all if it is on the `false` branch of a conditional. However, we must be doing something with scope resolution. My knowledge about that part is especially shaky as those parts of compilation is moving to dyno more and more everyday.

Reply:
I do see a difference of 5s during resolution adding/removing just this if statement with `--no-checks` (testing with Arkouda). In the interest of making this PR as lean as possible, I will restore the if statements and open an issue

Reply:
I've opened https://github.com/chapel-lang/chapel/issues/27519